include/cache.hpp

```diff
@@ -68,7 +68,7 @@ class fixed_sized_cache
 
     Value Get(const Key &key) const
     {
-        read_guard lock{safe_op};
+        write_guard lock{safe_op};
```

Comment:
why write_guard?

Reply:
`Get` calls `Touch`, which may rearrange elements in the `cache_policy` object.

An alternative that would require more code would be to add mutexes to each policy class that this affects (LRU and LFU), then write custom copy constructors for them. Might be worth it

Reply:
or you can pick between read and write guards, depending on the policy

Reply:
but implementing mutexes in policies is better